Sort by updated time feature for examples page

sinanata opened the pull request:
Attempted to add a simple **generate-timestamps** node script and a timestamp-based sort by update date feature to the examples page. I believe new features and updates are getting lost on newcomers who are not regularly following the project. I am not a developer, so please feel free to ignore if this is irrelevant. Cheers.
![image](https://github.com/user-attachments/assets/bf3bcc5a-4373-48e5-ac75-b38dc3b71ab8)

sinanata committed:
Sort by updated time feature for examples page

WestLangley commented:
Sorting by "date added" may make more sense than sorting by "date modified".

Or, just add the release ( e.g., "r177" ) as a tag to `tags.json`.

sinanata commented:
> Sorting by "date added" may make more sense than sorting by "date modified".
> 
> Or, just add the release ( e.g., "r177" ) as a tag to `tags.json`.

I got the created timestamp in timestamps.json but I thought updates might also be important for examples

cmhhelgeson commented:
I'd also prefer date added over date modified. Often, examples are trivially updated to use more up to date syntax, even if the core code of the example itself has not been updated. For instance, PRs like #28241 make minor modifications to a majority of examples in the codebase. If we were to go by date updated, then many samples that have been in the codebase for nearly a decade will seem to be at most a year old.

cmhhelgeson commented:
@Mugen87 I don't think this PR relates to documentation?

Mugen87 commented:
You're right, I've misinterpreted the screenshot. Deleted my comment to avoid confusion.

sinanata commented:
@cmhhelgeson got it, then I'll provide options and make the "added date" default, does that make sense?

sinanata committed:
minor fixes addressing requests from @cmhhelgeson

1- Added alphabetical and date created sorting, defaulting to date created
2- Now aligned with overall lowercase style
3- Removed "/" before "17 days ago" text which was out of context

sinanata committed:
Further simplification of the examples page sorting UI

sinanata commented:
WebGL and WebGPU separation seemed intentional, serving as main categories. In the currently live version, they are kept in search results.

If I go the "newest" path, I have to go for the "oldest", etc, too. I was aiming to reduce the number of words/icons from the feature.

mrdoob commented:
> As a newbie my first behavior was to check the examples, over a short period I came back at least 100 times to the examples page as I had a better idea of what I was looking for in various cases. 

Interesting yeah...

But it can be tricky because sometimes we modify many examples in bulk (like updating all of them to use import maps).

> Dynamic GitHub file URL linking for each example. With vibe-coding skyrocketing, you can’t believe the number of people who can’t make the example/github-repo connection instinctively

Do you mean a link to the github page for each example?

sinanata commented:
1- I'd keep dateAdded, but I need to see the frequency data of unique updates to see if making dateUpdated makes sense.

2- I see there are links only in the browser window; my behavior was wheel-clicking on relevant items to review, which unique pages don't have, so my bad. Still might be worth improving, though.

sinanata commented:
Yup, but when you wheel-click (intelli-click with mouse) on the thumbnails, you end up going to the source page., I thought they were missing https://threejs.org/examples/webgl_animation_keyframes.html
